src/lib/util/flag-resolver.test.ts

```diff
@@ -29,12 +29,11 @@ test('should produce UI flags with extra dynamic flags', () => {
 
 test('should use external resolver for dynamic flags', () => {
     const externalResolver = {
-        isEnabled: (name: string) => {
-            if (name === 'extraFlag') {
-                return true;
-            }
-        },
-        getVariant: () => defaultVariant,
+        isEnabled: (name: string) => name === 'extraFlag',
+        getVariant: (name: string) => ({
+            ...defaultVariant,
+            feature_enabled: name === 'extraFlag',
+        }),
```

Comment:
This change in the existing test seems meaningful because:
- `isEnabled` returns `true`
- but `feature_enabled` in the default variant is `false`

These two should always be the same, which was causing breakages in this test (due to invalid input data). I think this change makes sense and better reflects reality.